src/App.js

```diff
@@ -29,32 +66,260 @@ class App extends React.Component {
+
+  handleMessageUpdate = (key, updatedMessage) => {
+    const updatedMessages = this.state.messages.map((message) => {
+      if (message.key === key) {
+        return {
+          key: message.key,
+          val: { ...message.val, message: updatedMessage },
+        };
+      }
+
+      return message;
+    });
+
+    this.setState({ messages: updatedMessages });
+  };
+
+  likePost = (key) => {
+    const messageRef = ref(database, `${DB_MESSAGES_KEY}/${key}`);
+
+    runTransaction(messageRef, (currentData) => {
+      if (!currentData) {
+        return { numberOfLikes: 0 };
+      }
+
+      const currentLikes = currentData.numberOfLikes || 0;
+      if (currentLikes < 1) {
+        return { ...currentData, numberOfLikes: currentLikes + 1 };
+      } else {
+        return { ...currentData, numberOfLikes: currentLikes };
+      }
+    })
+      .then(() => {
+        // After successfully updating the likes in Firebase, fetch the updated data
+
+        const updatedMessageRef = ref(database, `${DB_MESSAGES_KEY}/${key}`); // reference to the message node in Firebase is created for that message with its unique key
+
+        return get(updatedMessageRef); // get() is called with that reference to retrieve the most recent data from Firebase at that message node
+      })
+      .then((snapshot) => {
+        const updatedMessage = { key: snapshot.key, val: snapshot.val() }; // creates an object containing the key and value of the updated message from the previous then method (snapshot)
+
+        const updatedMessages = this.state.messages.map((message) => {
+          // We now map through the current state's messages array to update the message we just modified in Firebase
+          if (message.key === key) {
+            // if message's key matches the key of our snapshot, we replace the specific message with the updatedMessage
+            return updatedMessage;
+          }
+
+          return message; // Else, we leave the message unchanged
+        });
+        this.setState({ messages: updatedMessages }); // we then set the state with the updated messages array that contains that 1 modified message object, updatedMessage
+      })
+      .catch((error) => {
+        console.error("Error updating likes: ", error);
+      });
+  };
+
+  logout = (e) => {
+    this.setState({
+      isLoggedIn: false,
+      user: {},
+    });
+
+    signOut(auth);
+
+    const { history } = this.props;
+    if (history) {
+      history.push("/");
+    }
   };
 
   render() {
-    // Convert messages in state to message JSX elements to render
-    let messageListItems = this.state.messages.map((message) => (
-      <li key={message.key}>{message.val}</li>
-    ));
+    console.log(this.state.messages);
+    console.log(this.state.user);
+
     return (
       <div className="App">
+        {this.state.isLoggedIn && this.state.user.email ? (
+          <NavBar loggedInUser={this.state.user.email} />
+        ) : (
+          <div></div>
+        )}
```

Comment:
```suggestion
        {this.state.isLoggedIn && this.state.user.email && (
          <NavBar loggedInUser={this.state.user.email} />
        )}
```

the empty div does not make sense. Either make it `null` or as suggested. My suggestion is a bit simpler though